king_phisher/client/dialogs/configuration.py

```diff
@@ -274,6 +279,31 @@ def interact(self):
 		self.dialog.destroy()
 		return response
 
+	def save_proxy_settings(self):
+		proxy_url = urllib.parse.urlparse(self.gtk_builder_get('entry_proxy_username').get_text())
+		proxy_username = self.gtk_builder_get('entry_proxy_username').get_text()
+		proxy_password = self.gtk_builder_get('entry_proxy_password').get_text()
+		if not proxy_url.hostname and not proxy_url.scheme:
+			gui_utilities.show_dialog_warning('Invalid Proxy Values', self.parent,
```

Comment:
This logic looks like it'll throw a warning if no proxy is specified. Since this is an optional setting we should only throw a warning if the value is incomplete. For example they specify `http://` (schema but no hostname) or they specify `test.com` (hostname but no schema).

The minimum things we would need would be the schema and hostname. Port can be inferred. You should also check that the schema is either `http` or `https`.